java/integTest/src/test/java/glide/modules/JsonTests.java

```diff
@@ -321,6 +321,276 @@ public void arrlen() {
         assertEquals(5L, res);
     }
 
+    @Test
+    @SneakyThrows
+    void test_json_numincrby() {
+        String key = UUID.randomUUID().toString();
+
+        var jsonValue =
+            "{"
+                + "    \"key1\": 1,"
+                + "    \"key2\": 3.5,"
+                + "    \"key3\": {\"nested_key\": {\"key1\": [4, 5]}},"
+                + "    \"key4\": [1, 2, 3],"
+                + "    \"key5\": 0,"
+                + "    \"key6\": \"hello\","
+                + "    \"key7\": null,"
+                + "    \"key8\": {\"nested_key\": {\"key1\": 69}},"
+                + "    \"key9\": 1.7976931348623157e308"
+                + "}";
+
+        // Set the initial JSON document at the key
+        assertEquals("OK", Json.set(client, key, "$", jsonValue).get());
+
+        // Test JSONPath
+        // Increment integer value (key1) by 5
+        Object result = Json.numincrby(client, key, "$.key1", 5).get();
+        assertArrayEquals(new Object[] {6L}, (Object[]) result); // Expect 1 + 5 = 6
+
+        // Increment float value (key2) by 2.5
+        result = Json.numincrby(client, key, "$.key2", 2.5).get();
+        assertArrayEquals(new Object[] {6.0}, (Object[]) result); // Expect 3.5 + 2.5 = 6
+
+        // Increment nested object (key3.nested_key.key1[0]) by 7
+        result = Json.numincrby(client, key, "$.key3.nested_key.key1[1]", 7).get();
+        assertArrayEquals(new Object[] {12L}, (Object[]) result); // Expect 4 + 7 = 12
+
+        // Increment array element (key4[1]) by 1
+        result = Json.numincrby(client, key, "$.key4[1]", 1).get();
+        assertArrayEquals(new Object[] {3L}, (Object[]) result); // Expect 2 + 1 = 3
+
+        // Increment zero value (key5) by 10.23 (float number)
+        result = Json.numincrby(client, key, "$.key5", 10.23).get();
+        assertArrayEquals(new Object[] {10.23}, (Object[]) result); // Expect 0 + 10.23 = 10.23
+
+        // Increment a string value (key6) by a number
+        result = Json.numincrby(client, key, "$.key6", 99).get();
+        assertArrayEquals(new Object[] {null}, (Object[]) result); // Expect null
+
+        // Increment a None value (key7) by a number
+        result = Json.numincrby(client, key, "$.key7", 51).get();
+        assertArrayEquals(new Object[] {null}, (Object[]) result); // Expect null
```

Comment:
note: a lot of these tests are testing the JSON module and not our client.  We don't need to test whether the JSON module is able to increment, we only need to test that our API can handle the return values appropriately.